src/components/PeopleTable.tsx

```diff
@@ -1,5 +1,64 @@
-/* eslint-disable jsx-a11y/control-has-associated-label */
-export const PeopleTable = () => {
+import { Person } from '../types/Person';
+import { Link, useParams, useSearchParams } from 'react-router-dom';
+import { useEffect, useState, useCallback } from 'react';
+import { SearchLink } from '../components/SearchLink';
+import cn from 'classnames';
+
+interface PeopleTableProps {
+  persons: Person[];
+  sortField: string | null;
+  sortOrder: 'asc' | 'desc' | null;
+}
+
+export const PeopleTable: React.FC<PeopleTableProps> = ({
+  persons,
+  sortField,
+  sortOrder,
+}) => {
+  const findPersonByName = (name: string | null) => {
+    return persons.find(p => p.name === name);
+  };
+
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+  const [searchParams] = useSearchParams();
+  const { slug } = useParams();
+
+  useEffect(() => {
+    const storedSlug = searchParams.get('selectedPerson');
+
+    if (storedSlug) {
+      const selectPerson = persons.find(person => person.slug === storedSlug);
+
+      setSelectedPerson(selectPerson || null);
+    }
+
+    if (slug) {
+      const selectedPersonSlug = persons.find(person => person.slug === slug);
+
+      setSelectedPerson(selectedPersonSlug || null);
+    }
```

Comment:
The logic here seems to handle both `storedSlug` and `slug`. However, if both are present, the `slug` will overwrite the `storedSlug` selection. Ensure this is the intended behavior.